django/utils/functional.py

```diff
@@ -440,3 +440,30 @@ def partition(predicate, values):
     for item in values:
         results[predicate(item)].append(item)
     return results
+
+
+class CallableBool:
+    """
+    An boolean-like object that is also callable for backwards compatibility.
+    """
+    def __init__(self, value):
+        self.value = value
+
+    def __bool__(self):
+        return self.value
+
+    def __call__(self):
+        warnings.warn(
+            "Using user.is_authenticated() and user.is_anonymous() as a method is deprecated.",
```

Comment:
So far those are indeed the only usecases, so you are right: it feels as though it should be moved to django.contribute.auth. That is was I did originally but my problem was: "how do I get the template engine to not emit a deprecation warning?" Importing CallableBool from django.contribute.auth led to an import loop. Suggestions?